yarn-project/aztec-rpc/src/aztec_rpc_server/aztec_rpc_server.ts

```diff
@@ -130,17 +153,34 @@ export class AztecRPCServer implements AztecRPC {
    * Retrieve the public key associated with an address.
    * Throws an error if the account is not found in the key store.
    *
-   * @param address - The AztecAddress instance representing the account.
+   * @param address - The AztecAddress instance representing the account to get public key for.
    * @returns A Promise resolving to the Point instance representing the public key.
    */
-  public async getAccountPublicKey(address: AztecAddress): Promise<Point> {
-    const result = await this.db.getPublicKey(address);
+  public async getPublicKey(address: AztecAddress): Promise<Point> {
+    const result = await this.db.getPublicKeyAndPartialAddress(address);
     if (!result) {
       throw new Error(`Unable to public key for address ${address.toString()}`);
     }
     return Promise.resolve(result[0]);
   }
 
+  /**
+   * Retrieve the public key and partial contract address associated with an address.
+   * Throws an error if the account is not found in the key store.
+   *
+   * @param address - The AztecAddress instance representing the account to get public key and partial address for.
+   * @returns A Promise resolving to the Point instance representing the public key.
+   */
+  public async getPublicKeyAndPartialAddress(
+    address: AztecAddress,
+  ): Promise<[Point, PartialContractAddress] | undefined> {
```

Comment:
```suggestion
  ): Promise<[Point, PartialContractAddress]> {
```

Since you are throwing if there's no result, right?

Reply:
Good catch. Addressed in [23a4f70](https://github.com/AztecProtocol/aztec-packages/pull/1103/commits/23a4f70e6562ac6e75faf1103945456332a72a43)